src/main/java/longah/node/Transaction.java

```diff
@@ -193,4 +198,43 @@ public void recalculateBalances() throws LongAhException{
         }
     }
 
+    /**
+     * Edits the specified transaction based on user input.
+     *
+     * @param expression   The user input for editing the transaction.
+     * @param memberList   The list of members in the group.
+     * @throws LongAhException If the transaction index is invalid or if the edit input is in an invalid format.
+     */
+    public void editTransaction(String expression, MemberList memberList) throws LongAhException {
+        // Reset the balances of all members involved in the transaction
+        this.lender.resetBalance();
+        for (Subtransaction subtransaction : this.subtransactions) {
+            subtransaction.getBorrower().resetBalance();
+        }
+        // Clear the existing subtransactions
+        subtransactions.clear();
+
+        // Parse the new transaction details
+        String[] splitInput = expression.split("p/");
+        if (splitInput.length < 2 || splitInput[0].isEmpty()) {
+            throw new LongAhException(ExceptionMessage.INVALID_EDIT_COMMAND);
+        }
+
+        String lenderName = splitInput[0].trim();
+        this.lender = memberList.getMember(lenderName);
+        double totalSumLent = 0.0;
+
+        for (int i = 1; i < splitInput.length; i++) {
+            String nameValue = splitInput[i].trim();
+            totalSumLent += addBorrower(nameValue, memberList);
+        }
+
+        // Replace the lender's balance with the new total sum lent
+        this.lender.addToBalance(totalSumLent);
+
+        // Update the borrowers' balances
+        updateBorrowerBalances();
```

Comment:
i will make the method to update this to reduce code repetition, kiv and accept for now